modules/aioseop_sitemap.php

```diff
@@ -4224,6 +4251,20 @@ public function get_all_post_type_data( $args ) {
 			}
 			$this->excludes = array_merge( $args['exclude'], $exclude_slugs ); // Add excluded slugs and IDs to class var.
 
+			if ( is_array( $aioseop_options['aiosp_cpostnoindex'] ) ) {
```

Comment:
@EkoJR this portion also does not support the aforementioned behavior. By removing all post types that have been marked as NOINDEX in general settings, you also remove posts that have been overridden to INDEX in post settings.

Reply:
Same as above

Reply:
@EkoJR can you please document each block of code that involves complex logic so that it is easy to understand what you are trying to do?

Reply:
I've added comments, not sure what else you want me to add though.

Reply:
@EkoJR this entire code block is doing something - what is it doing? what is the logic?  For example, in the `else` you are returning a number but not in the `if` without specifying this deviation. 

If you can include the logic either in the function doc block or in the code block (the latter being preferable), it would be easier to understand and review.

Reply:
@contactashish13 I did add comments...and I don't know what else I can add at this point.